pkg/crawl/identify.go

```diff
@@ -222,7 +222,11 @@ func (i *Log4jIdentifier) vulnerabilityFileWalkFunc(depth uint, result *Finding,
 			}
 		}
 
-		if strings.HasSuffix(path, ".class") {
+		filename, lastSlash := path, strings.LastIndex(path, "/")
```

Comment:
We are using exactly the same business logic in `pathMatchesLog4JVersion` to determine the filename from a path inside an archive.

Let's extract it to a named function.